### ISSUE TITLE
ArgparseController doesn't honor default for Python < 3

### ISSUE THREAD

**Issue by GatorQue** [datafolklabs/cement#474]
I would expect the default method to be called when using Python < 3 and no aliases in use for the following code:
```
from cement.core import backend
from cement.core.foundation import CementApp
from cement.ext.ext_argparse import ArgparseController, expose

# define an application base controller
class MyAppBaseController(ArgparseController):
    class Meta:
        label = 'base'
        description = "My Application does amazing things!"
        epilog = "This is the text at the bottom of --help."

        config_defaults = dict(
            foo='bar',
            some_other_option='my default value',
            )

        arguments = [
            (['-f', '--foo'],
             dict(action='store', help='the notorious foo option')),
            (['-C'],
             dict(action='store_true', help='the big c option')),
            ]

    @expose(hide=True)
    def default(self):
        self.app.log.info('Inside base.default function.')
        if self.app.pargs.foo:
            self.app.log.info("Recieved option 'foo' with value '%s'." % \
                          self.app.pargs.foo)

    @expose(help="this command does relatively nothing useful.")
    def command1(self):
        self.app.log.info("Inside base.command1 function.")

    @expose(help="more of nothing.")
    def command2(self):
        self.app.log.info("Inside base.command2 function.")


class MyApp(CementApp):
    class Meta:
        label = 'example'
        base_controller = MyAppBaseController


with MyApp() as app:
    app.run()
```

What happens when Python < 3 is used:
```
$ python cementapp.py
usage: example [-h] [--debug] [--quiet] [-f FOO] [-C]
               {command1,command2,default} ...
example: error: too few arguments
```

The default method is called when using CementBaseController with Python < 3.
```
from cement.core import backend
from cement.core.foundation import CementApp
from cement.core.controller import CementBaseController, expose

# define an application base controller
class MyAppBaseController(CementBaseController):
    class Meta:
        label = 'base'
        description = "My Application does amazing things!"
        epilog = "This is the text at the bottom of --help."

        config_defaults = dict(
            foo='bar',
            some_other_option='my default value',
            )

        arguments = [
            (['-f', '--foo'],
             dict(action='store', help='the notorious foo option')),
            (['-C'],
             dict(action='store_true', help='the big c option')),
            ]

    @expose(hide=True)
    def default(self):
        self.app.log.info('Inside base.default function.')
        if self.app.pargs.foo:
            self.app.log.info("Recieved option 'foo' with value '%s'." % \
                          self.app.pargs.foo)

    @expose(help="this command does relatively nothing useful.")
    def command1(self):
        self.app.log.info("Inside base.command1 function.")

    @expose(help="more of nothing.")
    def command2(self):
        self.app.log.info("Inside base.command2 function.")


class MyApp(CementApp):
    class Meta:
        label = 'example'
        base_controller = MyAppBaseController


with MyApp() as app:
    app.run()
```

When using the above with Python < 3:
```
$ python cementapp.py
INFO: Inside base.default function.
```

**Comment by derks** [comment]
@GatorQue thank you for your feedback!  This is listed as a limitation in the documentation for the argparse extension:

- http://builtoncement.com/2.10/api/ext/ext_argparse.html#requirements
-
 http://builtoncement.com/2.10/api/ext/ext_argparse.html#cement.ext.ext_argparse.ArgparseController.Meta.default_func

Notably:

> Some features of ArgparseController are only available in Python 3 including controller and function/command aliases (Python 3+) and controller default functions/command (Python 3.4+).


One of the intentional design choices with `ArgparseController` was to build off of `argparse` functionality directly with regard to subparsers and to support Cement's sub-command/sub-controller/etc.  A side-effect of that is what you've mentioned above in that Argparse itself does not allow for a `default` action in Python < 3 (atleast the way we are using it).

I think this is related:

- https://bugs.python.org/issue9253


The older `CementBaseController` is doing a lot nasty/hackish things outside of, and around argparse in order to support sub-commands/sub-controllers/etc... and doesn't even use argparse sub-parser features at all, which is why it will no longer be available in Cement 3.

**Comment by GatorQue** [comment]
As a developer I find it most helpful if an API provide identical functionality regardless of which version of Python their users might be using. I am still in the early stages of development so it hasn't yet become a problem for me and will try and take this into account in my applications design. Have you reached out to argparser to see if they would be willing to fix this for Python 2.7?
